src/core/edm/sagas/initializeOrganizationDataSet.js

```diff
@@ -0,0 +1,186 @@
+import { List, Map, fromJS } from 'immutable';
+import { Models, Types } from 'lattice';
+import {
+  AuthorizationsApiActions,
+  AuthorizationsApiSagas,
+  SearchApiActions,
+  SearchApiSagas,
+} from 'lattice-sagas';
+import { DataUtils, Logger } from 'lattice-utils';
+import type { Saga } from '@redux-saga/core';
+import type {
+  AccessCheck,
+  Organization,
+  PropertyType,
+  UUID,
+} from 'lattice';
+import type { WorkerResponse } from 'lattice-sagas';
+import type { SequenceAction } from 'redux-reqseq';
+
+import { toSagaError } from '../../../utils';
+import {
+  ERR_MISSING_ORG,
+  ERR_MISSING_PROPERTY_TYPE,
+  ERR_UNEXPECTED_SEARCH_RESULTS,
+} from '../../../utils/constants/errors';
+import {
+  DATA_SET,
+  DATA_SET_COLUMNS,
+  HITS,
+  IS_OWNER,
+} from '../../redux/constants';
+import { selectOrganization, selectPropertyTypes } from '../../redux/selectors';
+import { MAX_HITS_10000 } from '../../search/constants';
+import { INITIALIZE_ORGANIZATION_DATA_SET, initializeOrganizationDataSet } from '../actions';
+import { FQNS } from '../constants';
+
+const { AccessCheckBuilder, FQN } = Models;
+const { PermissionTypes } = Types;
+const { getAuthorizations } = AuthorizationsApiActions;
+const { getAuthorizationsWorker } = AuthorizationsApiSagas;
+const { searchEntitySetData } = SearchApiActions;
+const { searchEntitySetDataWorker } = SearchApiSagas;
+const { getPropertyValue } = DataUtils;
+
+const REQUIRED_PROPERTY_TYPES :FQN[] = [
+  FQNS.OL_DATA_SET_ID,
+  FQNS.OL_ID,
+];
+
+const LOG = new Logger('EDMSagas');
+
+function* initializeOrganizationDataSetWorker(action :SequenceAction) :Saga<WorkerResponse> {
+
+  let workerResponse :WorkerResponse;
+
+  try {
+    yield put(initializeOrganizationDataSet.request(action.id, action.value));
+
+    const {
+      dataSetId,
+      organizationId,
+    } :{|
+      dataSetId :UUID;
+      organizationId :UUID;
+    |} = action.value;
+
+    const organization :?Organization = yield select(selectOrganization(organizationId));
+    if (!organization) {
+      throw new Error(ERR_MISSING_ORG);
+    }
+
+    const propertyTypes :Map<UUID, PropertyType> = yield select(selectPropertyTypes(REQUIRED_PROPERTY_TYPES));
+    const propertyTypeIds :Map<FQN, UUID> = propertyTypes.map((propertyType) => propertyType.type).flip();
+    if (propertyTypeIds.count() !== REQUIRED_PROPERTY_TYPES.length) {
+      throw new Error(ERR_MISSING_PROPERTY_TYPE);
+    }
+
+    const accessChecks :AccessCheck[] = [
+      (new AccessCheckBuilder())
+        .setAclKey([dataSetId])
+        .setPermissions([PermissionTypes.OWNER])
+        .build()
+    ];
+    const authorizationsCall = call(getAuthorizationsWorker, getAuthorizations(accessChecks));
+
+    const dataSetSearchCall = call(
+      searchEntitySetDataWorker,
+      searchEntitySetData({
+        constraints: [{
+          constraints: [{
+            searchTerm: `entity.${propertyTypeIds.get(FQNS.OL_ID)}:${JSON.stringify(dataSetId)}`,
```

Comment:
No change, but I've seen this a couple of times. Maybe we add a a `getEntitySetSearchTerm` or a more general `getSearchTerm` to `lattice-utils`? 
```
const getSearchTerm = (propertyTypeId, value) => `entity.${propertyTypeId}:${JSON.stringify(value)}`;
```

Reply:
I agree, wrapping this is a good idea. I've thought about it a bit, just not sure how to do it. A generic `getSearchTerm` doesn't make sense because the format matters. `getEntitySetSearchTerm` makes more sense but that also feels a bit off. At the same time, something like `getPropertyTypeSearchTermToMatchExactValue` is pretty ridiculous 😆 .

Reply:
you could pass it an `exact :boolean` arg that would handle that in the `getEntitySetSearchTerm` fn.

Reply:
It took me way too long to figure this out, so we should definitely make this easier. From ES:

> The query string “mini-language” is used by the Query string and by the q query string parameter in the search API.

I think what we actually need is a query string builder. For example, this is what I'm currently struggling with:

```
constraints: [
  { searchTerm: `NOT(entity.${propertyTypeIds.get(FQNS.OL_FLAGS)}:audit) AND ${query}` },
],
```

Maybe something like this would be worth considering to bring in:

https://github.com/sudo-suhas/elastic-builder

Reply:
Actually, https://github.com/sudo-suhas/elastic-builder won't work for us because it seems to generate what ES needs, which is different than what our api needs.